src/llv8.cc

```diff
@@ -200,14 +200,65 @@ std::string LLV8::LoadString(int64_t addr, int64_t length, Error& err) {
   return res;
 }
 
+std::string LLV8::Utf16ToUtf8(const std::u16string& u16_str) {
+  if (u16_str.empty()) {
+    return std::string();
+  }
+  const char16_t* p = u16_str.data();
+  std::u16string::size_type len = u16_str.length();
+  if (p[0] == 0xFEFF) {
+    // bom
+    p += 1;
+    len -= 1;
+  }
+
+  std::string u8_str;
+  u8_str.reserve(len * 3);
+
+  char16_t u16_char;
+  for (std::u16string::size_type i = 0; i < len; ++i) {
+    u16_char = p[i];
+    if (u16_char < 0x0080) {
+      u8_str.push_back((char)(u16_char & 0x00FF));
+      continue;
+    }
+    if (u16_char >= 0x0080 && u16_char <= 0x07FF) {
+      u8_str.push_back((char)(((u16_char >> 6) & 0x1F) | 0xC0));
+      u8_str.push_back((char)((u16_char & 0x3F) | 0x80));
+      continue;
+    }
+    if (u16_char >= 0xD800 && u16_char <= 0xDBFF) {
+      uint32_t highSur = u16_char;
+      uint32_t lowSur = p[++i];
+      uint32_t codePoint = highSur - 0xD800;
+      codePoint <<= 10;
+      codePoint |= lowSur - 0xDC00;
+      codePoint += 0x10000;
+      u8_str.push_back((char)((codePoint >> 18) | 0xF0));
+      u8_str.push_back((char)(((codePoint >> 12) & 0x3F) | 0x80));
+      u8_str.push_back((char)(((codePoint >> 06) & 0x3F) | 0x80));
+      u8_str.push_back((char)((codePoint & 0x3F) | 0x80));
+      continue;
+    }
+    {
+      u8_str.push_back((char)(((u16_char >> 12) & 0x0F) | 0xE0));
+      u8_str.push_back((char)(((u16_char >> 6) & 0x3F) | 0x80));
+      u8_str.push_back((char)((u16_char & 0x3F) | 0x80));
+      continue;
+    }
+  }
+
+  return u8_str;
+}
```

Comment:
Can this be replaced by `ConvertUTF16toUTF8()` from llvm's `Support/ConvertUTF.h`?